File: 1920-build-array-from-permutation.py
"""
1920. Build Array from Permutation

Submitted: October 16, 2024

Runtime: 105 ms (beats 7.96%)
Memory: 16.70 MB (beats 82.92%)
"""

class Solution:
    def buildArray(self, nums: List[int]) -> List[int]:
        return [nums[nums[i]] for i in range(len(nums))]

File: 1920-build-array-from-permutation.cpp
/*
1920. Build Array from Permutation

Submitted: October 16, 2024

Runtime: 8 ms (beats 10.42%)
Memory: 19.02 MB (beats 52.24%)
*/

class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
        vector<int> permutated;
        permutated.reserve(nums.size());
        for (int i = 0; i < nums.size(); ++i) {
            permutated.push_back(nums.at(nums.at(i)));
        }
        return permutated;
    }
};